mirror-service-x/src/main/java/org/hyades/vulnmirror/datasource/osv/OsvMirrorConfiguration.java

```diff
@@ -0,0 +1,33 @@
+package org.hyades.vulnmirror.datasource.osv;
+
+import org.apache.commons.lang3.concurrent.BasicThreadFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+class OsvMirrorConfiguration {
+
+    @Produces
+    @ApplicationScoped
+    @Named("osvExecutorService")
+    ExecutorService executorService() {
+        final var threadFactory = new BasicThreadFactory.Builder()
+                .namingPattern("hyades-mirror-osv-%d")
+                .uncaughtExceptionHandler((thread, exception) -> {
+                    final Logger logger = LoggerFactory.getLogger(OsvMirror.class);
+                    logger.error("An uncaught exception was thrown while mirroring NVD", exception);
+                })
+                .build();
+
+        return new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<>(1), threadFactory);
```

Comment:
The queue likely needs to be larger than `1`, because we get one event per ecosystem.

With the queue having a capacity of `1`, the following would happen:

```
1. OSV:Maven  -> processing
2. OSV:Go     -> queued
3. OSV:Debian -> dropped
4. OSV:NPM    -> dropped
5. ...        -> dropped
```

So either the capacity needs to be increased to at least the number of ecosystems that exist (~50?), or the mirroring event for OSV needs to provide all to-be-mirrored ecosystems in a single event.

Reply:
changed this one to accept a comma separated list of ecosystems and parse them in a loop